langchain4j-core/src/main/java/dev/langchain4j/internal/ExceptionMapper.java

```diff
@@ -0,0 +1,69 @@
+package dev.langchain4j.internal;
+
+import dev.langchain4j.exception.AuthenticationException;
+import dev.langchain4j.exception.HttpException;
+import dev.langchain4j.exception.InternalServerException;
+import dev.langchain4j.exception.InvalidRequestException;
+import dev.langchain4j.exception.ModelNotFoundException;
+import dev.langchain4j.exception.RateLimitException;
+import dev.langchain4j.exception.TimeoutException;
+import dev.langchain4j.exception.UnrecoverableException;
+import dev.langchain4j.exception.UnresolvedModelServerException;
+
+import java.nio.channels.UnresolvedAddressException;
+import java.util.concurrent.Callable;
+
+@FunctionalInterface
+public interface ExceptionMapper {
+
+    ExceptionMapper DEFAULT = new DefaultImpl();
+
+    static <T> T mappingException(Callable<T> action) {
+        return DEFAULT.withExceptionMapper(action);
+    }
+
+    default <T> T withExceptionMapper(Callable<T> action) {
+        try {
+            return action.call();
+        } catch (Exception e) {
+            throw mapException(e);
+        }
+    }
+
+    RuntimeException mapException(Exception e);
+
+    class DefaultImpl implements ExceptionMapper {
```

Comment:
```suggestion
    class DefaultExceptionMapper implements ExceptionMapper {
```
nit: to be consistent with naming of all our other default implementations